CONTRIBUTING.md

```diff
@@ -158,3 +223,319 @@ including how to setup a local Kubernetes cluster, can be found in
+  * steps and/or configuration to reproduce the bug
+  * any tests that demonstrate the presence of the bug
+* please attach any relevent logs or diagnostic output
+
+#### Cleanup
+
+A `kind/cleanup` label categorizes an issue or PR as related to cleaning up
+code, process, or technical debt.
+
+To create a cleanup issue:
+
+* label your issue with `kind/cleanup`
+* describe your cleanup in the issue body being sure to include
+  * what is being cleaned
+  * for what reason it is being cleaned (technical debt, deprecation, etc.)
+
+Examples of a cleanup include:
+* Adding comments to describe code execution
+* Making code easier to read and follow
+* Removing dead code related to deprecated features or implementations
+
+#### Feature
+
+A `kind/feature` label categorizes an issue or PR as related to a new feature.
+
+To create a feature issue:
+
+* label your issue with `kind/feature`
+* describe your proposed feature in the issue body being sure to include
+  * a use case for the new feature
+  * list acceptance tests for the new feature
+  * describe any dependencies for the new feature
+* depending on the size and impact of the feature
+  * a design proposal may need to be submitted
+  * the feature may need to be discussed in the community meeting
+
+Before you begin work on your feature it is import to ensure that you have
+proper review and approval from the project maintainers.
+
+Examples of a new feature include:
+* Adding a new set of metrics for enabling additional telemetry.
+* Adding additional supported transport layer protocol options for network policy.
+* Adding support for IPSec.
+
+#### Deprecation
+
+A `kind/deprecation` label categorizes an issue or PR as related to feature
+marked for deprecation.
+
+To create a deprecation issue:
+
+* label your issue with `kind/deprecation`
+* title the issue with the feature you are deprecating
+* describe the deprecation in the issue body making sure to:
+  * explain why the feature is being deprecated
+  * discuss time-to-live for the feat
+  * discuss any impacts to existing APIs
+
+#### Design
+
+A `kind/design` label categorizes issue or PR as related to design.
+
+A design issue is for discussing larger architectural and design proposals.
+Approval of a design proposal may result in multiple additional feature,
+api-change, or cleanup issues being created to implement the design.
+
+To create a design issue:
+
+* label your issue with `kind/design`
+* describe the design in the issue body
+
+Before creating additional issues that implement the proposed design it is
+important to get feedback and approval from the maintainers. Design feedback
+could include some of the following:
+
+* needs additional detail
+* no, this problem should be solved in another way
+* this is desireable but we need help completing other issues first; then we will
+  consider this design
+
+_Note this was previously called `proposal`. We will add the `kind/design` label
```

Comment:
Should we not include this note in the PR itself, as it will become irrelevant as soon as this is merged? It also applies to other labels (`bug` -> `kind/bug`).